src/utilities/params.ts

```diff
@@ -84,30 +84,30 @@ const numberParam: ParamGetSet<unknown> = {
   },
 }
 
-export function getParamValue<T extends Param>(value: string, param: T): ExtractParamType<T>
-export function getParamValue<T extends Param>(value: string, param: T): unknown {
+export function getParamValue<T extends Param>(value: string | undefined, param: T): ExtractParamType<T>
+export function getParamValue<T extends Param>(value: string | undefined, param: T): unknown {
```

Comment:
this was always technically true, but when I updated `getParamValuesFromUrl` which used to return string[] to be `getParamValueFromUrl` which now returns `string | undefined`, it correctly was upset with me

Reply:
it's safe to use `!` for value below because we expect the `undefined` to mean the param was optional, which is handled by optional ParamGetSet. If somehow `undefined` gets used with param type like `String` or any non-optional, those params will handle it gracefully by throwing `invalid()`

Reply:
This seems suspicious to me. `ParamGet` says the value must be a string. For example `stringParam` accepts and returns the value passed in. But it expects it to be a string, and it does not call `invalid` if its not a string.

Reply:
I think you're right. I will revisit this